source/Octopus.Client/Util/ParameterHelper.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Repositories.Async;
+
+namespace Octopus.Client.Util
+{
+    static class ParameterHelper
+    {
+        public static Dictionary<string, object> CombineParameters(Dictionary<string, object> currentAdditionalQueryParameters, object userProvidedParameters)
+        {
+            if (currentAdditionalQueryParameters == null)
+                currentAdditionalQueryParameters = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+
+            if (userProvidedParameters == null)
+                userProvidedParameters = new { };
+            var resultDictionary = GetParameters(userProvidedParameters);
+
+            // Value from the current repository(currentAdditionalQueryParameters) overrides the one from user (userProvidedParameters)
+            currentAdditionalQueryParameters.ToList().ForEach(x => resultDictionary[x.Key] = x.Value);
+            return new Dictionary<string, object>(resultDictionary);
+        }
+
+        public static void ValidateSpaceParameters(SpaceQueryParameters curreSpaceQueryParameters, bool newIncludeGlobal,
```

Comment:
This method should probably take in two instances of `SpaceQueryParameters` and compare them. Feels a little strange that you are essentially passing the same type of data to this method in two different ways.